kolibri/utils/sanity_checks.py

```diff
@@ -103,3 +106,45 @@ def check_log_file_location():
         if os.path.exists(old_log_path):
             new_log_path = os.path.join(home, "logs", log_location_update[log])
             shutil.move(old_log_path, new_log_path)
+
+
+def migrate_databases():
+    """
+    Try to migrate all active databases. This should not be called unless Django has
+    been initialized.
+    """
+    from django.conf import settings
+
+    for database in settings.DATABASES:
+        call_command("migrate", interactive=False, database=database)
+
+    # load morango fixtures needed for certificate related operations
+    call_command("loaddata", "scopedefinitions")
+
+
+def check_database_is_migrated():
+    """
+    Use a check that the database instance id model is initialized to check if the database
+    is in a proper state to be used. This must only be run after django initialization.
+    """
+    apps.check_apps_ready()
+    from morango.models import InstanceIDModel
+
+    try:
+        InstanceIDModel.get_or_create_current_instance()[0]
+        return
+    except OperationalError:
+        try:
+            migrate_databases()
```

Comment:
It worries me that a sanity check performs operations? It's a "check", so this goes a bit beyond the scope. Also, now `cli` calls command `migrate_databases` inside `sanity_checks`, that seems to be a creep of module scope.

Suggesting this: Sanity checks throw exceptions. The caller handles the exceptions. This would mean throwing an exception in `sanity_checks` and handling it in `cli` by calling `migrate_databases`.

Reply:
Sorry about the hindsight comment.

Reply:
Following up here: https://github.com/learningequality/kolibri/pull/6580